csm/conf/setup.py

```diff
@@ -426,7 +426,7 @@ def _rsyslog_common(self):
             setup_info = self.get_data_from_provisioner_cli(const.GET_SETUP_INFO)
         except ProvisionerCliError as e:
             Log.warn(f"Salt command failed {e}")
-        if setup_info.get(const.STORAGE_TYPE) == const.STORAGE_TYPE_VIRTUAL:
+        if setup_info and setup_info.get(const.STORAGE_TYPE) == const.STORAGE_TYPE_VIRTUAL:
```

Comment:
Please create a common function to validate storage type with ignoring case. Same have been done for RESTART Check. Extract the code and create common function and use it evrerywhere

Reply:
I have updated code, extracted a common function to check if storage is virtual and used it. In case of unsupported feature could not use this.

Reply:
There is another call for get get_node_id  from provisioner it cannot be clubbed.